hawkular-command-gateway-war/pom.xml

```diff
@@ -72,11 +72,6 @@
     </dependency>
 
     <dependency>
-      <groupId>org.hawkular.inventory</groupId>
-      <artifactId>hawkular-inventory-json-helper</artifactId>
-    </dependency>
```

Comment:
> So what I would like to see rather than have one "general" class to handle all the forwarding, perhaps have a superclass that all command execution classes extend - this superclass would provide utility methods like "forwardToFeed" or "forwardToBus" but have (again, as an example) an abstract method called "obtainRecipientIDs(BasicMessage command)" that must be implemented by the command - the command would know "I am the ExecuteOperationByGroup" command, so I know I should look for the "groupPath" attribute, and from there I need to ask inventory what feeds manage the resources in that group".
>
> Or, as an alternative, perhaps we have some basic command design patterns that we look for. Maybe this one uber class would look for one of several types of addressing JSON elements. For example, maybe it says "is there a 'resourcePath' attribute? If so, use that to find the one feed ID as the recipient. If there is no resourcePath, is there a 'resourcePaths' attribute (notice the "s") - if so, use that to get the multiple feed IDs that manage those multiple resources. In the future we could add "does the JSON command have a "groupPath" attribute - if so, use that and determine the feeds that manage that group of resources"
> 
> In this way, you don't need to have these special "receipientId" things in the special AuthMessage. We just have a design pattern where each command must have one of the several different attributes that are supported (like "resourcePath" or "resourcePaths" or "groupPath"). Note that all of commands already have this, except for EchoCommand - which isn't targeting any feed today but should be (right now EchoCommand just echos from the server, it doesn't forward it to the agent so it isn't the agent that is getting pings via the Echo its just the server. But the Echo stuff is just for testing anyway. So we can do whatever for that.)

Yes, I am thinking of something like you propose. Generally, I am sticking to the assumption that 

    the request type determines routing.

Technically, the kind of routing will be determined by some of the supported routing interfaces (such as `ResourcePathDestination`, `GroupPathDestination` and `ResourcePathsDestination`) that a request will have to implement.

The routing command should then be chosen based on the routing interface. Let me try that.